Automattic/wp-calypso
# ccTLDs fr-form: Add VAT Validation

The backend vat validation has been tightened up, so we want to upgrade the client-side to match to prevent some unfortunate UX.

Notably, spaces in a VAT Id are not ok, as they cause errors with our 3rd party API, but we want to avoid a round trip to the server to tell the user to remove a space. Let's just remove it for them immediately instead.

Testing Instructions:

1. From your site: Domains > Add
2. Choose a .fr domain
3. Fill out the first page and hit "Continue"
4. Select "A company or organization" to reveal the VAT Number field if it's not already visible
5. Verify that the field is sanitized properly. The field should uppercase input, and allow only alphanumeric characters, `+` and `*` (both used in IE VAT Ids).

![fr-form-vat-sanitization](https://user-images.githubusercontent.com/5952255/32153640-91409b42-bd77-11e7-8f89-a615efdd9133.gif)
